libcontainer/integration/exec_test.go

```diff
@@ -1992,3 +1992,70 @@ next_fd:
 		t.Fatalf("found %d extra fds after container.Run", count)
 	}
 }
+
+// Test that a container using user namespaces is able to bind mount a folder
+// that has some part of its path with "---" permissions for others
+func TestBindMountAndUser(t *testing.T) {
+	if _, err := os.Stat("/proc/self/ns/user"); os.IsNotExist(err) {
+		t.Skip("userns is unsupported")
+	}
+	if testing.Short() {
+		return
+	}
+
+	rootfs, err := newRootfs()
+	ok(t, err)
+	defer remove(rootfs)
+
+	temphost, err := ioutil.TempDir("", "mnt1host")
+	ok(t, err)
+	defer os.RemoveAll(temphost)
```

Comment:
We have recently switched to go 1.15+, so please use `t.TempDir()` -- much cleaner that way.

Reply:
Oh, yeah, already did that when rebasing yesterday :)